server/jetstream.go

```diff
@@ -1265,21 +1267,27 @@ func (a *Account) EnableJetStream(limits map[string]JetStreamAccountLimits) erro
 	return nil
 }
 
-// Return whether or not we require MaxBytes to be set.
-func (a *Account) maxBytesRequired(cfg *StreamConfig) bool {
+// Return whether we require MaxBytes to be set and if > 0 an upper limit exists
+func (a *Account) maxBytesRequired(cfg *StreamConfig) (bool, int64) {
 	a.mu.RLock()
 	jsa := a.js
 	a.mu.RUnlock()
 	if jsa == nil {
-		return false
+		return false, 0
 	}
 	jsa.mu.RLock()
 	selectedLimits, _, ok := jsa.selectLimits(cfg)
 	jsa.mu.RUnlock()
 	if !ok {
-		return false
+		return false, 0
 	}
-	return selectedLimits.MaxBytesRequired
+	maxStreamBytes := int64(0)
+	if cfg.Storage == MemoryStorage {
+		maxStreamBytes = selectedLimits.MemoryMaxStreamBytes
```

Comment:
I see that in this PR, where we call maxBytesRequired() and "true" is returned, we still check for returned "limit > 0" in order to complain, so that is fine, but my first reaction was: what if we return true and those limits have not been configured?

Reply:
those two limits are independent of each other.
Do you think I should rename maxBytesRequired to maxBytesLimits?